modules/service/src/main/scala/lucuma/odb/graphql/mapping/EmailMapping.scala

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.odb.graphql
+
+package mapping
+
+import grackle.skunk.SkunkMapping
+
+import table.*
+
+trait EmailMapping[F[_]]
+  extends ProgramTable[F]
+  with EmailTable[F] {
+
+  lazy val EmailMapping = 
+    ObjectMapping(
+      tpe = EmailType,
+      fieldMappings = List(
+        SqlField("id", EmailTable.EmailId, key = true, hidden = true),
+        SqlField("senderEmail", EmailTable.SenderEmail),
+        SqlField("recipientEmail", EmailTable.RecipientEmail),
+        SqlField("subject", EmailTable.Subject),
+        SqlField("textMessage", EmailTable.TextMessage),
+        SqlField("htmlMessage", EmailTable.HtmlMessage),
+        SqlField("originalTime", EmailTable.OriginalTime),
+        SqlField("status", EmailTable.Status),
+        SqlField("statusTime", EmailTable.StatusTime),
+        // SqlObject("program", Join(EmailTable.ProgramId, ProgramTable.Id)) // do we want this here?
```

Comment:
Really, I only have the program id in t_email to facilitate sending programEdit subscription messages. It doesn't seem necessary to provide the program off of the Email, IMO.

Reply:
One reason, maybe, would be if we provide queries in the API to return say all the emails associated with programs matching a given `WherePrograms`.  Perhaps we don't plan on that though and in any case there's no reason to add it until we do.

Reply:
We might want to do something like that in the future. But, it is unclear to me at the moment. I foresee us sending out other types of emails in the future. We may want to provide a view of all emails, regardless of type, or maybe not. I'm guessing we will probably want to add an `EmailType` to the table in the future. At that point we can decide if/how we want to provide additional access to the emails.

Reply:
Also, we can add emails to the Programs in the API without providing the Program for an Email. I guess I never see emails having their own top level query, but who knows. Maybe we'll send emails that aren't program specific and we'll have to make the program id optional in the table?